### ISSUE TITLE
Support for Darwin builds

### ISSUE THREAD

**Issue by copumpkin** [lkl/linux#345]
Is LKL expected to work on macOS? If I try to build it, it fails quickly with:

```
Makefile:78: *** Unrecognized platform: .  Stop.
```

It seems like in principle it should be able to work just fine as a library, but I don't know if there are annoyances in practice. I see that https://github.com/lkl/linux/issues/18 suggests that compiling on LLVM could be difficult, and @tavip suggests that the linker scripts could make this difficult.

**Comment by Rondom** [comment]
Not familiar with Darwin at all: Is there anything that speaks against compiling with GCC?
For a start, have you tried starting with adding darwin support to arch/lkl/Makefile and trying to get at least vmlinux to compile?

**Comment by copumpkin** [comment]
Yeah, I can compile fairly easily with GCC. I haven't tried touching anything yet, but might try hacking it up a bit later and see if I can get the compile to go through. How screwed are we with the linker though?

**Comment by zokrezyl** [comment]
Had also a look into this. The challenges are following:
# build scripts
* ld-version.sh needs to be addapted to deal with macos LD (ld -v instead of ld --version)
* ld-version.sh should not fail 

# macho binary
* there is unfortunatelly no loader/executor for macho (macos) binary type

I think it would be great to have this. Porting certain apps to macos would be much easier I guess.
